### PR TITLE
windows: set job object for executor and children

### PR DESCRIPTION
**Pull request by tgross**
On Windows, if the `raw_exec` driver's executor exits, the child processes are not also killed. Create a Windows "job object" (not to be confused with a Nomad job) and add the executor to it. Child processes of the executor will inherit the job automatically. When the handle to the job object is freed (on executor exit), the job itself is destroyed and this causes all processes in that job to exit.

Fixes: https://github.com/hashicorp/nomad/issues/23668
Ref: https://learn.microsoft.com/en-us/windows/win32/procthread/job-objects

~Note: automated testing for this will need to be end-to-end but we're not running Windows on nightly anymore.~ I've verified the behavior works as expected using Process Explorer. Killing the executor kills all child processes as expected. Killing the root child process causes the executor to exit (as usual) and that in turn kills any child processes that didn't exit already.

_Edit: added tests, which requires changing what runs in CI as well_

### CODE REVIEW

**Comment by tgross** [comment]
As noted in sidebar conversation I'm going to get `drivers/shared` running on Windows and add a test for this there.

File: drivers/rawexec/driver_test.go [hunk 1]
```diff
@@ -237,103 +236,6 @@ func TestRawExecDriver_StartWait(t *testing.T) {
 	require.NoError(harness.DestroyTask(task.ID, true))
 }
 
-func TestRawExecDriver_StartWaitRecoverWaitStop(t *testing.T) {
```
Comment (tgross) [inline drivers/rawexec/driver_test.go:240]: Note to reviewers: this is moved to `driver_unix_test.go` unchanged, including any modernization to use `shoenig/test`, `t.Cleanup`, etc. I'll take a second pass over these files in a separate refactoring PR to do all that.

File: drivers/shared/executor/executor_test.go [hunk 2]
```diff
@@ -59,15 +59,6 @@ var (
 	compute  = topology.Compute()
 )
 
-type testExecCmd struct {
```
Comment (tgross) [inline drivers/shared/executor/executor_test.go:62]: Note to reviewers: this whole file is marked `!windows` now, so I moved these helpers into `utils_test.go`

File: drivers/rawexec/driver_windows_test.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+)
+
+// TestRawExecDriver_ExecutorKill verifies that killing the executor will stop
+// its child processes
+func TestRawExecDriver_ExecutorKill(t *testing.T) {
+	ci.Parallel(t)
+
+	d := newEnabledRawExecDriver(t)
+	harness := dtestutil.NewDriverHarness(t, d)
+	t.Cleanup(func() { harness.Kill() })
```
Comment (schmichael) [inline drivers/rawexec/driver_windows_test.go:28]:
```suggestion
	t.Cleanup(harness.Kill)
```

**Review by schmichael — APPROVED** [review]

**Pull request merged**